File: frontend/angular-ecommeerce/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItems } from "./order-items";

export class Purchase {

    customer: Customer;
    shippingAddress: Address;
    billingAddress: Address;
    order: Order;
    orderItems: OrderItems[];
}

File: backend/src/main/java/com/example/ecommerceprojectsite/dto/Purchase.java
package com.example.ecommerceprojectsite.dto;

import com.example.ecommerceprojectsite.entities.Address;
import com.example.ecommerceprojectsite.entities.Customer;
import com.example.ecommerceprojectsite.entities.Order;
import com.example.ecommerceprojectsite.entities.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;

}
